internal/rest/resources/cluster.go

```diff
@@ -590,14 +544,27 @@ func clusterMemberDelete(s *state.State, r *http.Request) response.Response {
 		return response.SmartError(err)
 	}
 
+	// Run the PostRemove hook locally.
 	err = state.PostRemoveHook(s, force)
 	if err != nil {
 		return response.SmartError(err)
 	}
 
+	// Run the PostRemove hook on all other members.
+	remotes := s.Remotes()
 	err = cluster.Query(s.Context, true, func(ctx context.Context, c *client.Client) error {
 		c.SetClusterNotification()
-		return c.DeleteClusterMember(ctx, name, force)
+		addrPort, err := types.ParseAddrPort(c.URL().URL.Host)
+		if err != nil {
+			return err
+		}
+
+		remote := remotes.RemoteByAddress(addrPort)
+		if remote == nil {
+			return fmt.Errorf("No remote found at address %q run the post-remove hook", c.URL().URL.Host)
+		}
+
+		return internalClient.RunHook(ctx, &c.Client, remote.Name, internalTypes.PostRemove, internalTypes.HookConfig{Force: force})
```

Comment:
As this is now calling the hook on each member, is there a situation where the actual API endpoint needs to forward the request to another member? Please see my comment from before for reference.